connectors/dynamodb-connector/src/main/scala/zio/connect/dynamodb/DynamoDBConnector.scala

```diff
@@ -0,0 +1,54 @@
+package zio.connect.dynamodb
+
+import software.amazon.awssdk.services.dynamodb.model.ResourceNotFoundException
+import zio._
+import zio.aws.core.{AwsError, GenericAwsError}
+import zio.aws.dynamodb.model._
+import zio.aws.dynamodb.model.primitives.{AttributeName, TableName}
+import zio.stream.{ZSink, ZStream}
+
+trait DynamoDBConnector {
+
+  def batchGetItem(request: => BatchGetItemRequest)(implicit
```

Comment:
This should be a sink consuming requests. And because we want the responses we can produce a chunk with the results when the sink is done.

```
def batchGetItem:
ZSink[Any, AwsError, BatchGetItemRequest, BatchGetItemRequest, Chunk[BatchGetItemResponse]]
```

Same with batchWriteItem, getItem, query, scan. 

Operations like listTables are different because they actually produce streams inherently. It's the stream of the tables.

Reply:
Certainly can make these changes, but I partially disagree. I'll document my reasoning here in case this gets revisited.

- `scan` and `query` are `ZStreams` by construction in the underlying `zio-aws-dynamodb` API; see their implementations below. It would be odd to me that the API considers these operations as streams and this connector converts them to sinks:

```scala
  override def query(request: => QueryRequest): ZStream[Any, AwsError, Map[AttributeName, AttributeValue]] =
    db.query(request).map(_.view.mapValues(_.asEditable).toMap)

  override def scan(request: => ScanRequest): ZStream[Any, AwsError, Map[AttributeName, AttributeValue]] =
    db.scan(request).map(_.view.mapValues(_.asEditable).toMap)
}
```

For the `getItem` and `batchGetItem` operations, I can see how they make sense as sinks, my one concern, however, is that the thing you get back after unwrapping them from the response object `Map[AttributeName, AttributeValue]` is something that is likely going to require transformation anyway similar to the decoding step for a stream of bytes, but maybe this is not relevant.

Reply:
Okay updated, current ci failure is due to couchbase connector test